pg_shard.c

```diff
@@ -530,17 +533,36 @@ ExtractRangeTableEntryWalker(Node *node, List **rangeTableList)
 
 /*
  * DistributedQueryShardList prunes the shards for the table in the query based
- * on the query's restriction qualifiers, and returns this list.
+ * on the query's restriction qualifiers, and returns this list. If the function
+ * cannot find any shards for the distributed table, it errors out. In other sense,
+ * the function errors out or returns a non-empty list.
  */
 static List *
 DistributedQueryShardList(Query *query)
 {
-	Oid distributedTableId = ExtractFirstDistributedTableId(query);
+	List *restrictClauseList = NIL;
+	List *prunedShardList = NIL;
 
-	List *restrictClauseList = QueryRestrictList(query);
+	Oid distributedTableId = ExtractFirstDistributedTableId(query);
 	List *shardIntervalList = LookupShardIntervalList(distributedTableId);
-	List *prunedShardList = PruneShardList(distributedTableId, restrictClauseList,
-										   shardIntervalList);
+
+	/* error out if no shards exists for the table */
+	if (shardIntervalList == NIL)
+	{
+		char *relName = get_rel_name(distributedTableId);
```

Comment:
We use `relationName` for relation names in our code (PostgreSQL uses `relname`, no uppercase, but we prefer the full word).